examples/mean-reversion.go

```diff
@@ -0,0 +1,241 @@
+	}
+
+	// Figure out when the market will close so we can prepare to sell beforehand.
+	clock, _ := alp.client.GetClock()
+	timeToClose := int((clock.NextClose.UnixNano() - clock.Timestamp.UnixNano()) / 1000000)
+	if timeToClose < 60000*15 {
+		// Close all positions when 15 minutes til market close.
+		fmt.Println("Market closing soon.  Closing positions.")
+
+		positions, _ := alp.client.ListPositions()
+		for _, position := range positions {
+			var orderSide string
+			if position.Side == "long" {
+				orderSide = "sell"
+			} else {
+				orderSide = "buy"
+			}
+			qty, _ := position.Qty.Float64()
+			qty = math.Abs(qty)
+			alp.submitMarketOrder(int(qty), position.Symbol, orderSide)
+		}
+		// Run script again after market close for next trading day.
+		fmt.Println("Sleeping until market close (15 minutes).")
+		time.Sleep((60000 * 15) * time.Millisecond)
+	} else {
+		// Rebalance the portfolio.
+		alp.rebalance()
+		time.Sleep(60000 * time.Millisecond)
+	}
+}
+
+// Spin until the market is open.
+func (alp alpacaClientContainer) awaitMarketOpen() bool {
+	clock, _ := alp.client.GetClock()
+	if clock.IsOpen {
+		return true
+	}
+	timeToOpen := int(((clock.NextOpen.UnixNano() - clock.Timestamp.UnixNano()) / 1000000000.0) / 60.0)
+	fmt.Printf("%d minutes til next market open.\n", timeToOpen)
+	return false
+}
+
+// Rebalance our position after an update.
+func (alp alpacaClientContainer) rebalance() {
+	// Get our position, if any.
+	positionQty := 0
+	positionVal := 0.0
+	position, err := alp.client.GetPosition(alpacaClient.stock)
+	if err != nil {
+	} else {
+		positionQty = int(position.Qty.IntPart())
+		positionVal, _ = position.MarketValue.Float64()
+	}
+
+	// Get the new updated price and running average.
+	amtBars := 20
+	bars, _ := alp.client.GetSymbolBars(alpacaClient.stock, alpaca.ListBarParams{Timeframe: "minute", Limit: &amtBars})
+	currPrice := float64(bars[len(bars)-1].Close)
+	alpacaClient.runningAverage = 0.0
+	for _, bar := range bars {
+		alpacaClient.runningAverage += float64(bar.Close)
+	}
+	alpacaClient.runningAverage /= 20.0
+
+	if currPrice > alpacaClient.runningAverage {
+		// Sell our position if the price is above the running average, if any.
+		if positionQty > 0 {
+			fmt.Println("Setting position to zero")
+			alp.submitLimitOrder(positionQty, alpacaClient.stock, currPrice, "sell")
+		} else {
+			fmt.Println("No position in the stock.  No action required.")
+		}
+	} else if currPrice < alpacaClient.runningAverage {
+		// Determine optimal amount of shares based on portfolio and market data.
+		account, _ := alp.client.GetAccount()
+		buyingPower, _ := account.BuyingPower.Float64()
+		positions, _ := alp.client.ListPositions()
+		portfolioVal, _ := account.Cash.Float64()
+		for _, position := range positions {
+			rawVal, _ := position.MarketValue.Float64()
+			portfolioVal += rawVal
+		}
+		portfolioShare := (alpacaClient.runningAverage - currPrice) / currPrice * 200
```

Comment:
What is 200?

Reply:
That's directly from the C# algo, it's the strat that they used.